src/scales/scale.time.js

```diff
@@ -184,86 +170,182 @@ function parse(scale, input) {
 }
 
 /**
- * Figures out what unit results in an appropriate number of auto-generated ticks
+ * Returns the number of unit to skip to be able to display up to `capacity` number of ticks
+ * in `unit` for the given `min` / `max` range and respecting the interval steps constraints.
  */
-function determineUnitForAutoTicks(minUnit, min, max, capacity) {
-	const ilen = UNITS.length;
-	let i, interval, factor;
+function determineStepSize(scale, unit, capacity) {
+	const {_adapter: adapter, min, max} = scale;
+	const count = adapter.diff(max, min, unit);
+	const interval = INTERVALS[unit];
+	const steps = interval.steps;
+	const target = Math.max(Math.floor(count / Math.max(1, capacity)), 1);
+
+	if (!steps) {
+		return target;
+	}
+
+	let i = 0;
+	let maxi = steps.length - 1;
```

Comment:
nit: It looks a little weird to me to have max and i combined without camel casing. maybe just max? maxI could be another option